build.py errors for Python 3.6 on Windows

This error came up by running the tests from this repo as part of the IDAES/idaes-pse integration tests. Here's the error log:

```txt
============================= test session starts =============================
platform win32 -- Python 3.6.8, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: D:\a\idaes-pse\idaes-pse\examples-pse, configfile: pytest.ini
plugins: cov-2.10.1
collected 1 item / 2 errors

=================================== ERRORS ====================================
_________________ ERROR collecting tests/test_build_script.py _________________
tests\test_build_script.py:18: in <module>
    import build
build.py:120: in <module>
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
E   AttributeError: module 'asyncio' has no attribute 'WindowsSelectorEventLoopPolicy'
------------------------------- Captured stdout -------------------------------
path: ['D:\\a\\idaes-pse\\idaes-pse\\examples-pse', 'D:\\a\\idaes-pse\\idaes-pse\\examples-pse\\tests', 'C:\\hostedtoolcache\\windows\\Python\\3.6.8\\x64\\Scripts\\pytest.exe', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\python36.zip', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\DLLs', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages\\win32', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages\\win32\\lib', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages\\Pythonwin']
__________________ ERROR collecting tests/test_notebooks.py ___________________
tests\test_notebooks.py:17: in <module>
    import build
build.py:120: in <module>
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
E   AttributeError: module 'asyncio' has no attribute 'WindowsSelectorEventLoopPolicy'
```

. It seems to be limited to the specific combination Python 3.6 and Windows, and wasn't caught by the tests here in examples-pse because the CI is not running on the full Python version and OS test matrix just yet (which will be rectified ASAP).

I think the fix shouldn't be too hard, essentially just add a special case within the special case in build.py#L117-120.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -33,7 +33,7 @@ jobs:
       run: |
         echo $PWD
         ls -lh .
-        python -m install --progress-bar=off -r requirements.txt
+        python -m pip install --progress-bar=off -r requirements.txt
     - name: Install IDAES
       run: |
         python -m pip install --progress-bar=off git+https://github.com/idaes/idaes-pse

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -44,7 +44,19 @@ jobs:
         idaes --version
     - name: Install IDAES extensions
       run: |
-        idaes get-extensions --platform ubuntu1804
+        idaes get-extensions --verbose
+        # NOTE: these commands are Unix-shell specific 
+        # and will need to be changed if we want to test on CMD.exe/Powershell
+        find $(idaes data-directory) -ls
+        # add bin directory to $PATH (only valid for subsequent steps)
+        echo "$(idaes bin-directory)" >> $GITHUB_PATH
+    - name: Test access to executables
+      run: |
+        if [ "$RUNNER_OS" == "Windows" ]; then
+            ipopt.exe -v
+        else
+            ipopt -v
+        fi
     - name: Run non-integration tests
       run: |
         pytest --verbose -m "not integration" tests/

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -2,36 +2,48 @@ name: Run examples
 
 on:
   push:
+    branches:
+      - main
   pull_request:
+    branches:
+      - main
   workflow_dispatch:
 
 jobs:
   examples:
-    runs-on: ubuntu-18.04
+    name: Build examples (py${{ matrix.python-version }}/${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # if fail-fast == true (the default), jobs for the remaining values in the matrix are cancelled
+      fail-fast: false
+      matrix:
+        python-version:
+          - '3.6'
+          - '3.7'
+        os:
+          - ubuntu-18.04
+          - windows-latest
     steps:
     - uses: actions/checkout@v2
-    - name: Setup Conda env
-      # `conda activate` does not work properly because env vars are not persisted between steps
-      # to circumvent this, instead of creating and activating a new Conda env,
-      # we install the packages in the Conda base environment, which is active by default
-      # caveat: specify full path to executables in the $CONDA/bin directory
-      run: |
-        $CONDA/bin/conda install --quiet --yes --name base python=3.7 pip
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
     - name: Install own dependencies
       run: |
         echo $PWD
         ls -lh .
-        $CONDA/bin/pip install --progress-bar=off -r requirements.txt
+        python -m install --progress-bar=off -r requirements.txt
     - name: Install IDAES
       run: |
-        $CONDA/bin/pip install --progress-bar=off git+https://github.com/idaes/idaes-pse
-        $CONDA/bin/idaes --version
+        python -m pip install --progress-bar=off git+https://github.com/idaes/idaes-pse
+        idaes --version
     - name: Install IDAES extensions
       run: |
-        $CONDA/bin/idaes get-extensions --platform ubuntu1804
+        idaes get-extensions --platform ubuntu1804
     - name: Run non-integration tests
       run: |
-        $CONDA/bin/pytest --verbose -m "not integration" tests/
+        pytest --verbose -m "not integration" tests/
     - name: Save Sphinx build artifacts
       uses: actions/upload-artifact@v2
       with:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -9,6 +9,10 @@ on:
       - main
   workflow_dispatch:
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
   examples:
     name: Build examples (py${{ matrix.python-version }}/${{ matrix.os }})

--- a/build.py
+++ b/build.py
@@ -114,8 +114,10 @@ _log.setLevel(logging.INFO)
 
 
 # This is a workaround for a bug in some versions of Tornado on Windows for Python 3.8
-# the sys.version_info check includes the micro version, so e.g. 3.7.1 will also evaluate to True
-if sys.platform == "win32" and sys.version_info > (3, 7):
+# this is the recommended fix for this according to e.g. https://github.com/tornadoweb/tornado#2608
+# it should be revisited with python>=3.9 or if a fix is implemented by e.g. Jupyter
+# the sys.version_info check includes the micro version, so e.g. 3.8.0 will also evaluate to True
+if sys.platform == "win32" and sys.version_info > (3, 8):
     import asyncio
 
     asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

--- a/build.py
+++ b/build.py
@@ -114,7 +114,8 @@ _log.setLevel(logging.INFO)
 
 
 # This is a workaround for a bug in some versions of Tornado on Windows for Python 3.8
-if sys.platform == "win32":
+# the sys.version_info check includes the micro version, so e.g. 3.7.1 will also evaluate to True
+if sys.platform == "win32" and sys.version_info > (3, 7):
     import asyncio
 
     asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

